### ISSUE TITLE
Update to 6.2 from 6.1: 

### ISSUE THREAD

**Issue by DonRichards** [IQSS/dataverse#10907]
**What steps does it take to reproduce the issue?**

* When does this issue occur?
When navigating to a dataset from a dataverse collection.

* Which page(s) does it occurs on?
Clicking on one of the URLs to a dataset from this page (https://example.com/dataverse/root?q=&types=dataverses%3Adatasets&sort=dateSort&order=desc&page=5) to (https://example.com/dataset.xhtml?persistentId=doi:10.7281/T1/YZKNJ6) results in a 500 error. 

* What happens?

```
HTTP Status 500 - Internal Server Error
type Exception report
message Internal Server Error
description The server encountered an internal error that prevented it from fulfilling this request.
exception
jakarta.servlet.ServletException: /dataset.xhtml @595,164 rendered="#{settingsWrapper.makeDataCountDisplayEnabled and DatasetPage.doi}": The class 'edu.harvard.iq.dataverse.DatasetPage' does not have the property 'doi'.
root cause
jakarta.el.PropertyNotFoundException: /dataset.xhtml @595,164 rendered="#{settingsWrapper.makeDataCountDisplayEnabled and DatasetPage.doi}": The class 'edu.harvard.iq.dataverse.DatasetPage' does not have the property 'doi'.
root cause
jakarta.el.PropertyNotFoundException: The class 'edu.harvard.iq.dataverse.DatasetPage' does not have the property 'doi'.
note The full stack traces of the exception and its root causes are available in the Payara Server 6.2023.8 #badassfish logs.
Payara Server 6.2023.8 #badassfish
```

* To whom does it occur (all users, curators, superusers)?
Not sure.

* What did you expect to happen?
Not throw an error

**Which version of Dataverse are you using?**
6.2

**Any related open or closed issues to this bug report?**
I don't believe so.

**Screenshots:**

<img width="1728" alt="Screenshot 2024-10-03 at 5 49 59 PM" src="https://github.com/user-attachments/assets/cac3c766-fdf7-4243-a6a1-40de7263a701">

**Are you thinking about creating a pull request for this issue?**
Not sure how to approach this.

**Stack trace**
This captures the essence of the issue: a `PropertyNotFoundException` is thrown because the `DatasetPage` class does not have a `doi` property, which is causing a rendering issue in the `dataset.xhtml` view. I trimmed unnecessary details, hopefully retaining key information for diagnosis.
```
[2024-10-03T17:51:50.675-0400] [Payara 6.2023.8] [SEVERE] [] [jakarta.enterprise.resource.webcontainer.faces.application] [tid: _ThreadID=96] 
Error Rendering View[/dataset.xhtml]
jakarta.el.PropertyNotFoundException: /dataset.xhtml @595,164 rendered="#{settingsWrapper.makeDataCountDisplayEnabled and DatasetPage.doi}": 
The class 'edu.harvard.iq.dataverse.DatasetPage' does not have the property 'doi'.
	at com.sun.faces.facelets.el.TagValueExpression.getValue(TagValueExpression.java:75)
	at jakarta.faces.component.UIComponentBase.isRendered(UIComponentBase.java:309)
	at com.sun.faces.renderkit.html_basic.HtmlBasicRenderer.encodeRecursive(HtmlBasicRenderer.java:250)
	at jakarta.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:557)
	...
Caused by: jakarta.el.PropertyNotFoundException: The class 'edu.harvard.iq.dataverse.DatasetPage' does not have the property 'doi'.
	at jakarta.el.BeanELResolver.getBeanProperty(BeanELResolver.java:620)
	...
```

**Comment by qqmyers** [comment]
FWIW: I'm not sure what is going on. DatasetPage in v6.2 does have a doi property - due to the method at https://github.com/IQSS/dataverse/blob/a218417e875a41e03495581fb6f0e8207c07bd75/src/main/java/edu/harvard/iq/dataverse/DatasetPage.java#L6436.

**Comment by pdurbin** [comment]
@DonRichards It looks like you're missing the fix at 7137ce9 which was part of 6.3.

It was reported in this issue:

- #10462

And fixed in this PR:

- #10463

**Comment by DonRichards** [comment]
That fixed it. Thanks
